source/detect-version.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+const inquirer = require('inquirer');
+const version = require('./version');
+const git = require('./git-util');
+const getCommits = require('./get-commits');
+const printCommitLog = require('./print-commit-log');
+
+const listCommits = commits => {
+	return commits.map(commit => `- ${commit.message}  ${commit.id}`);
+};
+
+const generateReleaseNotes = (commits, repoUrl, latest, nextTag) => {
+	const output = [];
+
+	const majorChanges = commits.filter(commit => commit.type === 'major');
+	const minorChanges = commits.filter(commit => commit.type === 'minor');
+	const patchChanges = commits.filter(commit => commit.type === 'patch');
+
+	if (majorChanges.length > 0) {
+		output.push('## Breaking changes', '', ...listCommits(majorChanges), '');
+	}
+
+	if (minorChanges.length > 0) {
+		output.push('## Features', '', ...listCommits(minorChanges), '');
+	}
+
+	if (patchChanges.length > 0) {
+		output.push('## Bug fixes', '', ...listCommits(patchChanges), '');
+	}
+
+	output.push('## All changes', '', `${repoUrl}/compare/${latest}...${nextTag}`);
+
+	return output.join('\n');
+};
+
+// Ask user about each commit and automatically determine the next release version
+module.exports = async (options, pkg) => {
+	const commits = await getCommits();
+
+	if (commits.length === 0) {
+		const answers = await inquirer.prompt([{
+			type: 'confirm',
+			name: 'confirm',
+			message: 'No commits found since previous release, continue?',
+			default: false
+		}]);
+
+		if (!answers.confirm) {
+			return {
+				...options,
+				...answers
+			};
+		}
+	}
+
+	await printCommitLog(commits, options.repoUrl);
+
+	const prompts = commits.map(commit => ({
+		type: 'list',
+		name: commit.id,
+		message: `What kind of change does "${commit.message}" commit contain?`,
```

Comment:
The commit message should be linkified so we can click it to see what it's about if the message is not clear.